zipline/algorithm.py

```diff
@@ -986,6 +995,15 @@ def set_commission(self, commission):
             raise OverrideCommissionPostInit()
         self.commission = commission
 
+    @api_method
+    def set_symbol_lookup_date(self, dt):
+        """
+        Set the date for which symbols will be resolved to their sids
+        (symbols may map to different firms or underlying assets at
+        different times)
+        """
+        self.symbol_lookup_date = dt
```

Comment:
I thought about this. My initial reaction was to the variable flow through to here: https://github.com/quantopian/zipline/blob/master/zipline/assets/assets.py#L324-L325, but a more explicit try-except would probably be more useful. Will update.